File: old/test/hazard2.cc
#include "script.h"
#include "state.h"

struct Hazard : public state_t {
    void go();
};


void Hazard::go()
{
    stack[0] = NULL;
    executed = 0;

    // Put 1 into [0], 254 into [1] and then flip between them.
    XOR(A);
    LOAD(X,A);
    INC(A);
    LOAD(Y,A);
    STA(X);
    LOAD(A,254);
    STA(Y);
    LOADM(A,X);
    LOADM(A,Y);
    LOADM(A,X);
    LOADM(A,Y);
    RET();
}


int main(int argc, char * argv[])
{
    sim_main(Hazard(), argc, argv);
}

File: src/hazard2.rs
use instructions::constants::*;

pub mod disassemble;
pub mod emitter;
pub mod instructions;
pub mod resistors;
pub mod spice_check;
pub mod spice_load;
pub mod state;

fn main() {
    let mut hazard = instructions::Instructions::default();
    hazard
        .xor  (A)
        .load (X, A)
        .inc  (A)
        .load (Y, A)
        .sta  (X)
        .load (A, 254)
        .sta  (Y)
        .load (A, [X])
        .load (A, [Y])
        .load (A, [X])
        .load (A, [Y])
    
        .ret();

    spice_check::spice_check_args(|_| hazard);
}